server/src/spacon/components/mqtt/core.clj

```diff
@@ -61,7 +82,7 @@
     (log/debugf "Connecting MQTT Client to %s" url)
     (try
       (do
-        (reset! conn (mh/connect url client-id))
+        (reset! conn (mh/connect url client-id mqtt-connect-ops))
```

Comment:
Also, we should allow connections to brokers that don't require any credentials.  So refactor this to set credentials if they exist, otherwise connect with `(mh/connect url client-id)`